File: artisanat-backend/src/main/java/com/artisanat_backend/dto/response/MediaResponseDto.java
package com.artisanat_backend.dto.response;

import com.artisanat_backend.enums.Type;
import com.artisanat_backend.model.Product;
import com.artisanat_backend.model.User;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class MediaResponseDto {
    private Long id;
    private String mediaUrl;
    private String mediaId;
    private Type type;
    private Product product;
    private User user;
}

File: artisanat-frontend/src/app/core/dtos/response/media-response-dto.ts
import {Type} from "../../enums/type.enum";
import {Product} from "../../models/product.model";
import {User} from "../../models/user.model";

export interface MediaResponseDto {
  id: number;
  mediaUrl: string;
  mediaId: string;
  type: Type;
  product: Product;
  user: User;
}
